state/factory/workingset.go

```diff
@@ -171,8 +171,12 @@ func (ws *workingSet) runAction(
 		return nil, errors.Wrapf(err, "Failed to get hash")
 	}
 	defer ws.ResetSnapshots()
-	if err := ws.freshAccountConversion(ctx, &actCtx); err != nil {
-		return nil, err
+
+	fCtx := protocol.MustGetFeatureCtx(ctx)
+	if fCtx.UseZeroNonceForFreshAccount && !fCtx.RefactorFreshAccountConversion && actCtx.Nonce == 0 {
```

Comment:
moving the condition outside for 2 purpose
1. make it clear it is only needed between these 2 hard-forks
2. for most cases, this `if` won't hit and avoid calling the func